tensorboard/plugins/interactive_inference/witwidget/notebook/visualization.py

```diff
@@ -499,6 +499,27 @@ def set_compare_custom_predict_fn(self, predict_fn):
       self.set_compare_model_name('2')
     return self
 
+  def set_custom_distance_fn(self, distance_fn):
+    """Sets a custom function for distance computation.
+
+    WIT can directly use a custom function for all distance computations within
+    the tool. In this case, the provided function should accept a query example
+    proto and a list of example protos to compute the distance against and
+    return a 1D list of numbers containing the distances.
+
+    Args:
+      distance_fn: The python function which will be used for distance
+      computation.
+
+    Returns:
+      self, in order to enabled method chaining.
+    """
+    if distance_fn is None:
+      self.delete('custom_distance_fn')
+    else:
+      self.store('custom_distance_fn', distance_fn)
+    return self
+
```

Comment:
can you update the smiling demo to use some sort of simple distance measure from the image pixels? or a pre-trained tf.js model that gives image embeddings :)

Reply:
I am deferring this in the interest of time.